docs/source/release_notes.rst

```diff
@@ -11,6 +11,7 @@
         * Update maintainers to add Frank :pr:`3382`
         * Allow woodwork version 0.14.0 to be installed :pr:`3381`
     * Documentation Changes
+        * Added section of AutoML Algorithms :pr:`3394`
```

Comment:
Could this be more specific? i.e. `Added automl user guide section covering search algorithms` or something along those lines.